test/cpp/tensorexpr/test_cuda.cpp

```diff
@@ -574,6 +574,226 @@ void testCudaNoThreadIdxWrite_1() {
   cudaFree(b_dev);
 }
 
+void testCudaSharedMemReduce_1() {
+  KernelScope kernel_scope;
+  // This test does the following:
+  //  for k in 0..1:  // block-idx
+  //    alloc(c, 64)
+  //    for n in 0..64:  // thread-idx
+  //      c(n) = 0
```

Comment:
For correctness I'd suggest also include `Free` in the test since `Allocate` is present.

Note for myself: IREval should report error if we have unfreed memory in the end of evalutation.